### ISSUE TITLE
I can't filter with a variable

### ISSUE THREAD

**Issue by rosostolato** [jin-qu/jinqu-odata#18]
## jinqu-odata version
v0.3.1

## Steps to reproduce
```
query.createQuery(DomeFeature).expand('users', ['enable'], u => u.userMail === user.mail)
```

## Expected  behavior
expected query: `?$expand=users($filter=userMail eq email@email.com;$select=enable)`

## Actual behavior
actual query: `?$expand=users($filter=userMail eq user/mail;$select=enable)`

**Comment by umutozel** [comment]
Thanks for the feedback.

Because we have to stringify the function and parse the expression to convert it to query string parameters, scope is lost (not like c# expressions). To overcome this JavaScript limitation, jinqu supports scope parameters.

```
query.createQuery(DomeFeature).expand('users', ['enable'], u => u.userMail === user.mail, { user })
```

This way, **user** variable will be available when parsing expression.

**Comment by rosostolato** [comment]
HI, thank you for the support!

I tried it and still building the same query:
`$expand=users($filter=userMail eq user/mail;$select=enable)`

I searched on jinqu docs and found something like this:
`const filtered = orders.where('c => c.id > value', { value: 3 })`

So I tested with this code and it didn't work either:
`query.createQuery(DomeFeature).expand('users', ['enable'], 'u => u.userMail === value', { value: user.mail })`


I tried to debug inner functions and inside `createExpandArgs` function, the scope is an array of array of object... maybe it can help you

**Comment by rosostolato** [comment]
This is what i'm talking about:
![debug](https://user-images.githubusercontent.com/27731641/65623013-4317bf00-df9d-11e9-8158-702b662baf54.png)

when the code finds `name in s`, s is an array with indexes 0, 1, 2... not an object

**Comment by umutozel** [comment]
I reproduced the issue, and will be working on it.
Jinqu is not resolving cascading properties for parameters, for now you can use it like this:

```javascript
const mail = user.mail;
query
   .createQuery(DomeFeature)
   .expand('users', ['enable'], u => u.userMail === mail, {mail})
```

**Comment by umutozel** [comment]
Fixed with 0.3.2 npm.
Thank you again for reporting.
